expression/builtin_string.go

```diff
@@ -1241,22 +1241,21 @@ func (c *hexFunctionClass) getFunction(ctx context.Context, args []Expression) (
 		return nil, errors.Trace(err)
 	}
 
-	switch t := args[0].GetTypeClass(); t {
-	case types.ClassString:
+	argTp := fieldTp2EvalTp(args[0].GetType())
+	switch argTp {
+	case tpString, tpDatetime, tpTimestamp, tpDuration, tpJSON:
```

Comment:
do we need a function like IsStringEvalTp

Reply:
no need for now